### PR TITLE
[MM-56732] Update keycloak docker configs and add make command

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
Using the latest image of keycloak as part of our local environment. Add it to your local start-up by adding `keycloak` to `ENABLED_DOCKER_SERVICES` in config.override.mk.

Also added, `make config-saml` in order to setup your config.json to work with your local keycloak

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56732

#### Release Note

```release-note
Update keycloak docker configs and add make command
```

### CODE REVIEW

File: server/build/docker/keycloak/README.md [hunk 1]
```diff
@@ -18,44 +18,12 @@ To use this keycloak image, we suggest you to use this configuration settings:
 - First Name Attribute: `firstName`
 - Last Name Attribute: `lastName`
```
Comment (hanzei) [inline server/build/docker/keycloak/README.md:19]: Should `make config-saml` also set these fields so it's easier to set up the connection between Mattermost and Keycloak?

Comment (hanzei) [inline server/build/docker/keycloak/README.md:19]: From my testing, this seems wrong. The settings that worked for me are
```suggestion
- First Name Attribute: `givenName`
- Last Name Attribute: `surname`
```

Can you please double-check what the correct setting is?

File: server/build/docker/keycloak/README.md [hunk 2]
```diff
@@ -18,44 +18,12 @@ To use this keycloak image, we suggest you to use this configuration settings:
...
 
-or overwrite your SamleSettings section with this settings in your config.json file (if you are not using
-database configuration) and restart the server:
+or overwrite your SamleSettings section in your config.json file by running `make config-saml` and restarting your server.
```
Comment (hanzei) [inline server/build/docker/keycloak/README.md:21]: Should `make config-saml` be the only way to configure the local dev server? i.e. do we even need to spell out the config value here?

File: server/build/docker/keycloak/README.md [hunk 3]
```diff
@@ -18,44 +18,12 @@ To use this keycloak image, we suggest you to use this configuration settings:
...
-        "LoginButtonTextColor": "#ffffff"
-    },
-```
+Admin Login:
+admin/admin
+
+Users:
+homer/password
+marge/password
+lisa/password
```
Comment (hanzei) [inline server/build/docker/keycloak/README.md:29]: Formating is a bit off 
```suggestion
Admin Login:
- admin/admin

Users:
- homer/password
- marge/password
- lisa/password
```

**Review by hanzei — CHANGES_REQUESTED** [review]
Awesome :rocket: This will make developing and debugging with SAML so much easier.

Should we add a short sub page in https://developers.mattermost.com/contribute/more-info/ to let people how that this exists? Maybe linking to https://github.com/mattermost/mattermost/blob/MM-56732/server/build/docker/keycloak/README.md is even enough.

File: server/scripts/mirror-docker-images.json [hunk 4]
```diff
@@ -16,7 +16,8 @@
     "1.4.0": "osixia/openldap:1.4.0@sha256:d5b2f2b816b25a1b57033b34f5d48c91cc3161a7d041811a9032604030bad9db"
   },
   "keycloak": {
-    "10.0.2": "jboss/keycloak:10.0.2@sha256:3720b5ace316b5790a58ce838f46e8cd44cedbdb7e35d3866311ddc5a5e71466"
+    "10.0.2": "jboss/keycloak:10.0.2@sha256:3720b5ace316b5790a58ce838f46e8cd44cedbdb7e35d3866311ddc5a5e71466",
```
[the remark this answers is not in the document]
Comment (BenCookie95) [reply server/scripts/mirror-docker-images.json:19]: Yeah I was asked to leave it by @mvitale1989 for now

File: server/build/docker/keycloak/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (BenCookie95) [reply server/build/docker/keycloak/README.md:21]: Yeah I'll make that clear here in the docs

File: server/build/docker/keycloak/README.md (hunk 1 above)
Comment (BenCookie95) [reply server/build/docker/keycloak/README.md:19]: I can double check

Comment (BenCookie95) [reply server/build/docker/keycloak/README.md:19]: The only problem is ldap has attributes of the same name so those would also get overwritten. I'll put the correct values in this README so people know

Comment (BenCookie95) [reply server/build/docker/keycloak/README.md:19]: Yeah those are correct, added to the README

**Comment by BenCookie95** [comment]
> Awesome 🚀 This will make developing and debugging with SAML so much easier.
> 
> Should we add a short sub page in https://developers.mattermost.com/contribute/more-info/ to let people how that this exists? Maybe linking to https://github.com/mattermost/mattermost/blob/MM-56732/server/build/docker/keycloak/README.md is even enough.

Thanks, I'll find a suitable place in the docs

File: server/build/docker/keycloak/README.md
Comment (hanzei) [inline server/build/docker/keycloak/README.md:1]: It would be nice to have some kind of introduction on what this directly does.

**Review by hanzei — APPROVED** [review]
LGTM

File: server/Makefile [hunk 5]
```diff
@@ -675,6 +675,21 @@ config-ldap: ## Configures LDAP.
 	@sed -i'' -e 's|"GroupDisplayNameAttribute": ".*"|"GroupDisplayNameAttribute": "cn"|g' ../config/config.json
 	@sed -i'' -e 's|"GroupIdAttribute": ".*"|"GroupIdAttribute": "entryUUID"|g' ../config/config.json
 
+config-saml: ## Configures SAML.
+	@echo Setting up configuration for local SAML with keycloak, please ensure your keycloak is running on http://localhost:8484
+
+	@cp build/docker/keycloak/keycloak.crt ../config/saml-idp.crt
+
+	@sed -i'' -e 's|"Verify": true|"Verify": false|g' ../config/config.json
+	@sed -i'' -e 's|"Encrypt": true|"Encrypt": false|g' ../config/config.json
```
[the remark this answers is not in the document]
Comment (BenCookie95) [reply server/Makefile:684]: Yeah good point, I can do that

**Comment by DHaussermann** [comment]
@vish9812 just an update on this. I have the tooling provided by Colton running with the working AD setup and users that are provided.

I now need to try upgrading the container to a Docker image that contains your code. Can you please resolve conflicts and merge `master` into this branch for me so that a new image is created.

**Comment by DHaussermann** [comment]
Sorry, I commented on the wrong Keycloak PR. Comment deleted for clarity.

**Review by DHaussermann — APPROVED** [review]
I did not fully test this as my local had an Okta setup that would interfere with this.  

I did however start the environment with the keycloak container and test the setup command. 

I will come back to this when I have a clean local available to test that the users provided can login. For now, we're fine to merge this. 

Huge thanks @BenCookie95 for providing this and helping to fill the gaps we have with SAML testing 🎉

**Comment by cwarnermm** [comment]
@BenCookie95 - Are there product documentation updates needed from your perspective for this PR?

**Pull request merged**